code/game/machinery/iv_drip.dm

```diff
@@ -287,39 +275,38 @@
 		var/list/options = list(
 			"IV drip" = image('icons/mob/screen/radial.dmi', "iv_drip"),
 			"Breath mask" = image('icons/mob/screen/radial.dmi', "iv_mask"))
-		var/chosen_action = show_radial_menu(usr, src, options, require_near = TRUE, radius = 42, tooltips = FALSE)
+		var/chosen_action = show_radial_menu(usr, src, options, require_near = TRUE, radius = 42, tooltips = TRUE)
 		if(!chosen_action)
 			return
 		switch(chosen_action)
 			if("IV drip")
 				if(attached)
-					visible_message("[usr] detaches [attached]'s [vein.name] from \the [src].")
+					visible_message("[usr] detaches \the [src] from [attached]'s [vein.name].")
 					vein = null
 					attached = null
 					blood_message_sent = FALSE
 					update_icon()
 					return
-				else
-					attached = over_object
-					vein = attached.get_organ(usr.zone_sel.selecting)
-					if(!vein)
-						to_chat(usr, SPAN_WARNING("They don't have that limb!"))
-						attached = null
-						vein = null
-						return
-					if(vein.status & ORGAN_ROBOT)
-						if(vein.robotize_type == PROSTHETIC_SYNTHSKIN)
-							to_chat(usr, SPAN_WARNING("The needle can't pierce through [attached]'s [vein.name] skin!"))
-							attached = null
-							vein = null
-							return
-						to_chat(usr, SPAN_NOTICE("You can't insert the needle in [attached]'s [vein.name]."))
+				attached = over_object
+				vein = attached.get_organ(usr.zone_sel.selecting)
+				if(!vein)
+					to_chat(usr, SPAN_WARNING("They don't have that limb!"))
+					attached = null
+					vein = null
+					return
+				if(vein.status & ORGAN_ROBOT)
```

Comment:
You should also check if they are wearing a voidsuit / armour / whatever, like syringes do. And maybe something in process for the same incase they put armour on while they have an IV needle in.

Reply:
The latter sounds like a nightmare that it's one of the few things I'm looking forward to be ordered to do, while the former I'm on the fence on since armor/voidsuit checks can be The Thing that either kills the patient or not, especially if they're wearing a RIG/hardsuit. Unless I can come up with a good solution to this, I'm not coding in that check

Reply:
Try not to mark things as resolved if they're not actually resolved, so I / other reviewers / maintainers can easily see them.

Reply:
I see. My bad about the resolved thing.

I'll see what I can do about this thing in particular. Ideally I really do not want to make cases with RIGs even more impossible to treat and just stare at them die. I'll probably just add an injection timer thing and make the IVs upgradable via Science so they can retain their bypass

Reply:
You could ask a maintainer about what their preferred solution might be.
A little timer while you "find an injection port" could work?

Reply:
Oh yeah, the armor check is already coded and working. I've been staring at the returns for the breather thing for the majority of today